File: python-algorithms/selection_sort.py
def find_smaller(arr):
    small = arr[0]
    small_index = 0
    for i in range(1, len(arr)):
        if arr[i] < small:
            small = arr[i]
            small_index = i
    return small_index
    
def selection_sort(arr):
    newArr = []
    for i in range(len(arr)):
        small_index = find_smaller(arr)
        newArr.append(arr.pop(small_index))
    return newArr
    
print(selection_sort([2, 4, 1, 6, 4, 12, 5]))

File: scripts/selectionSort.js
const findSmaller = (arr) => {
    let small = arr[0]
    let smallIndex = 0

    for (i in arr) {
        if(arr[i] < small) {
            small = arr[i]
            smallIndex = i
        }
    }
    return smallIndex 
}

const selectionSort = (arr) => {
    findSmaller(arr)
}

console.log(selectionSort([3, 5, 1, 6, 7, 12, 4 , 6]))
